### PR TITLE
`Development`: Automatically format JavaDoc comments via Spotless

### PR DESCRIPTION
**Pull request by Strohgelaender**
#### General
- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue
- [ ] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context
JavaDoc comments are currently not automatically formatted. This leads to inconsistencies across the code base and some PRs having changes to JavaDoc-Comments unrelated to the actually new implemented code, simply because the developer applied the IntelliJ auto-formatting. See e.g. #6043 in the `ProgrammingExerciseService` class.

### Description
This PR activates JavaDoc formatting via Spotless. From now on, any changed or new JavaDoc comment will be automatically formatted.

### Steps for Testing
scroll through the code

#### Code Review
- [x] Review 1
- [x] Review 2

### CODE REVIEW

**Review by simon-hng — APPROVED** [review]
fine_addition

File: src/main/java/de/tum/in/www1/artemis/config/CacheConfiguration.java [hunk 1]
```diff
@@ -86,6 +86,7 @@ public CacheManager cacheManager(HazelcastInstance hazelcastInstance) {
 
     /**
      * Setup the hazelcast instance based on the given jHipster properties and the enabled spring profiles.
+     *
```
Comment (b-fein) [inline src/main/java/de/tum/in/www1/artemis/config/CacheConfiguration.java:89]:
```suggestion
     *
```
A trailing whitespace is automatically added to the empty lines. Can this be removed somehow?

**Review by b-fein — CHANGES_REQUESTED** [review]

File: src/main/java/de/tum/in/www1/artemis/config/CacheConfiguration.java (hunk 1 above)
Comment (Strohgelaender) [reply src/main/java/de/tum/in/www1/artemis/config/CacheConfiguration.java:89]: Great catch. This seems to be an unresolved issue in the eclipse formatter, see https://stackoverflow.com/questions/15755671/how-to-stop-eclipse-formatter-from-adding-trailing-whitespaces-in-javadoc. Here the only solution is to run this as an additional taks, but nothing integrated into the formatter. I also tried updating the formatter version, but this is blocked due to https://github.com/diffplug/spotless/issues/1046

So I told IntellIJ to reformat everything, which removed that spaces. This isn't the best solution, but the only one I could come up with.

**Review by tobias-lippert — DISMISSED** [review]
nice improvement

**Review by ge65cer — APPROVED** [review]

**Review by julian-christl — APPROVED** [review]
Great QoL change, config changes look reasonable.
I did not check autogenerated spotless changes line by line for obvious reasons. They look good overall, though.

**Review by julian-christl — APPROVED** [review]

**Pull request merged**